phases/stack.go

```diff
@@ -1,9 +1,91 @@
 package phases
 
-import "fmt"
+import (
+	"io/ioutil"
+
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/go-utils/log"
+	"gopkg.in/yaml.v2"
+)
+
+var defaultStacks map[string]string = map[string]string{
+	"xamarin":      "osx-vs4mac-stable",
+	"cordova":      "osx-vs4mac-stable",
+	"react-native": "osx-vs4mac-stable",
+	"ionic":        "osx-vs4mac-stable",
+	"flutter":      "osx-vs4mac-stable",
+	"android":      "linux-docker-android",
+	"macos":        "osx-xcode-10.0",
+	"ios":          "osx-xcode-10.0",
+	"other":        "",
+}
+
+var optionsStacks = []string{
+	"linux-docker-android-lts",
+	"linux-docker-android",
+	"osx-vs4mac-beta",
+	"osx-vs4mac-previous-stable",
+	"osx-vs4mac-stable",
+	"osx-xamarin-stable",
+	"osx-xcode-10.0.x",
+	"osx-xcode-10.1.x",
+	"osx-xcode-10.2.x",
+	"osx-xcode-8.3.x",
+	"osx-xcode-9.2.x",
+	"osx-xcode-9.4.x",
+	"osx-xcode-edge",
+}
 
 // Stack ...
-func Stack() (string, error) {
-	fmt.Println("SetStack")
-	return "", nil
+func Stack(bitriseYMLPath string) (string, error) {
+
+	var stack string
+
+	(&option{
+		title:        "Choose stack selection mode",
```

Comment:
I think if your logic is able to check stack from bitrise.yml automatically then I would automatically skip this question. (or ask if this is the one he wants to use, or if not: then ask for selecting a stack. Also if not able to get the stack info.